addons/stock/models/stock_inventory.py

```diff
@@ -211,28 +160,77 @@ def action_cancel_draft(self):
         })
 
     def action_start(self):
-        for inventory in self.filtered(lambda x: x.state not in ('done','cancel')):
-            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
-            if (inventory.filter != 'partial') and not inventory.line_ids:
-                vals.update({'line_ids': [(0, 0, line_values) for line_values in inventory._get_inventory_lines_values()]})
-            inventory.write(vals)
-        return True
+        for inventory in self:
+            # Confirms the inventory adjustment and generates its lines if inventory
+            # state is draft
+            if inventory.state == 'draft':
+                vals = {
+                    'state': 'confirm',
+                    'date': fields.Datetime.now()
+                }
+                if not inventory.line_ids and not inventory.start_empty:
```

Comment:
are you sur you need the guard "if not inventory.line_ids" here?

Reply:
It's to avoid issues with potential duplicate lines when starting an inventory adjustment who have already some inventory lines. It can happen with demo data.

Reply:
maybe we should fix the demo data, like setting the state as confirm if it already added the inventory lines?